packages/shared_preferences/shared_preferences/lib/src/devtools_extension.dart

```diff
@@ -0,0 +1,136 @@
+// Copyright 2013 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'dart:convert';
+import 'dart:developer' as developer;
+
+import 'package:flutter/foundation.dart';
+
+import '../shared_preferences.dart';
+
+const String _eventPrefix = 'shared_preferences:';
+
+/// A typedef for the post event function
+@visibleForTesting
+typedef PostEvent = void Function(
+  String eventKind,
+  Map<String, Object?> eventData,
+);
+
+/// A helper class that provides data to the devtool extension.
+///
+/// It is only visible for testing and eval.
+@visibleForTesting
+class DevtoolsExtension {
```

Comment:
can we rename this `DevToolsExtensionDataProvider` or `SharedPreferencesDevToolsExtensionData` or similar? This class name is very close to `DevToolsExtension`, which is the wrapper widget from `package:devtools_extensions`, which is a bit confusing.